### PR TITLE
add new script for delete old index in elastic, update mariabackup and update of discusage 

### PR DESCRIPTION
**Pull request by falleri**

### CODE REVIEW

**Review by henrik-andreasson — CHANGES_REQUESTED** [review]
split it up in one pr per feature, eg sc_40 mariabackup and please remove the rpm-qa line

**Review by henrik-andreasson — APPROVED** [review]

**Review by henrik-andreasson — CHANGES_REQUESTED** [review]
remove rpm row

File: lang/939.english [hunk 1]
```diff
@@ -0,0 +1,5 @@
+HELP="Clean old backups, so the disk dont fill up"
+CLEANBAK_DESCR_1="Truncate index run successfully"
```
Comment (henrik-andreasson) [inline lang/939.english:2]: update names

**Review by henrik-andreasson — CHANGES_REQUESTED** [review]
update names this it not clean backups

**Review by henrik-andreasson — APPROVED** [review]

**Review by henrik-andreasson — APPROVED** [review]

File: lang/939.english (hunk 1 above)
Comment (falleri) [reply lang/939.english:2]: have update this

**Comment by falleri** [comment]
need to know if I need to do any more.

**Review by henrik-andreasson — APPROVED** [review]

File: related-available/938_mariabackup.sh [hunk 2]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+#cat ${DATAFILE} |cut -f3 -d"="| sort |uniq
+# Displays a list of valid backup directorys
+
+# Set the prompt for the select command
+PS3="Type a number or 'q' to quit, or press return to main menu: "
+
+# Create a list of files to display
+#fileList=$(stat -c '%n %.19y ' /backup/mariabackup/* )
+#fileList=$( cat ${DATAFILE} |cut -f3 -d"="| sort |uniq )
+fileList=$( cat /tmp/file.txt )
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:291]: use mktemp

File: related-available/938_mariabackup.sh [hunk 3]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+# Show a menu and ask for input. If the user entered a valid choice,
+# then invoke the editor on that file
+#select fileName in $fileList; do
+OIFS=$IFS
+IFS="
+"
+#select fileName in `cat ${DATAFILE} |cut -f3 -d"="| sort |uniq`; do
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:298]: remove comments if the code is done

File: related-available/938_mariabackup.sh [hunk 4]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+select fileName in `cat /tmp/file.txt`; do
+IFS=$OIFS
+    if [ -n "$fileName" ]; then
+        YESNO=Y
+        echo " Choose dir to use: ${fileName}"
+IFS="
+"
+	BCKFILE=$(basename $(echo "${fileName}"|awk '{print $1}'))
+        echo $BCKFILE
+        #select CA in `egrep "${fileName}" ${DATAFILE}|sort`;do
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:309]: remove commented code

File: related-available/938_mariabackup.sh [hunk 5]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+echo "ACTION: ${ACTION} TYPE ${TYPE}" 
+################ ACTION SWITCH, little messy ################################
 
-################# ACTION SWITCH ################################
-
+DATESTR=$(date +${DATESTING_FULL})
 if [ "${ACTION}" == "BACKUP" -a "${TYPE}" == "FULL" ] ; then
+# Full backup is only  possible once per date eq 2021-01-22
+  MARIABACKUP_DIR=${MARIABACKUP_BASEDIR}/${DATESTR}/${TYPE}
+  #backupdir=$(mariabackup_full_backup "${MARIABACKUP_DIR}")
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:345]: remove commented code

File: related-available/938_mariabackup.sh [hunk 6]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+  mariabackup_full_backup "${MARIABACKUP_DIR}"
+
+elif [  "${ACTION}" == "BACKUP" -a "${TYPE}" == "INC" ]; then
+# find last fullbackup find dir FULL, if not exist, exit
+# Need to have one fullbackup to use as base. with date, it is base  for all Incrementel backup
+# Try first with this day
+  DATESTR=$(ls -dtr /backup/mariabackup/*/FULL |tail -1 |cut -f4 -d"/")
+  #if [ -d ${MARIABACKUP_BASEDIR}/${DATESTR}/FULL ];then
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:353]: remove commented code

File: related-available/938_mariabackup.sh [hunk 7]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
-elif [  "${ACTION}" == "PREPARE" -a  "${TYPE}" == "INCREMENTAL" ] ; then
-
-  result=$(mariabackup_prepare_incremental)
+elif [  "${ACTION}" == "BACKUP" -a "${TYPE}" == "CUM" ] ; then
+
+# find last fullbackup find dir FULL, if not exist, exit
+# Try first with this day
+  DATESTR=$(ls -dtr /backup/mariabackup/*/FULL |tail -1 |cut -f4 -d"/")
+  if [ ! -z "${DATESTR}" ];then
+  #if [ -d ${MARIABACKUP_BASEDIR}/${DATESTR}/FULL ];then
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:383]: remove commented code

File: related-available/938_mariabackup.sh [hunk 8]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+       cd  ${MARIABACKUP_BASEDIR}/${DATESTR}/
+       #INDEXB=$(ls -trd ${MARIABACKUP_BASEDIR}/${DATESTR}/*_${TYPE} |tail -1 |cut -f1 -d"_")
+       INDEXB=$(ls -trd *_${TYPE} |tail -1 |cut -f1 -d"_")
+       INDEX=$(( $INDEXB + 1))
+       MARIABACKUP_DIR=${MARIABACKUP_BASEDIR}/${DATESTR}/${INDEX}_${TYPE}
+       BASEBCK=${MARIABACKUP_BASEDIR}/${DATESTR}/FULL
+    fi 
+    backupdir=$(mariabackup_incremental_backup "${BASEBCK}" "${MARIABACKUP_DIR}")
+
+ #    echo "mariabackup_incremental_backup ${BASEBCK} ${MARIABACKUP_DIR}"
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:399]: remove commented code

File: lang/938.english [hunk 9]
```diff
@@ -1,81 +1,81 @@
...
+HELP[2]=" no action needed"
 
-DESCR[3]="(%s)"
-HELP[3]=""
+DESCR[3]="Could not cretate backup"
+HELP[3]="Check space on device"
 
-DESCR[3]="(%s)"
-HELP[3]=""
+DESCR[4]="Incrementel backup ok"
```
Comment (henrik-andreasson) [inline lang/938.english:14]: spelling

File: lang/938.english [hunk 10]
```diff
@@ -1,81 +1,81 @@
...
+HELP[4]="Check space on device"
 
-DESCR[3]="(%s)"
-HELP[3]=""
+DESCR[5]="Desc 5"
```
Comment (henrik-andreasson) [inline lang/938.english:17]: ???

File: lang/938.english [hunk 11]
```diff
@@ -1,81 +1,81 @@
...
+HELP[5]="Check space on device"
 
-DESCR[3]="(%s)"
-HELP[3]=""
+DESCR[6]="Desc 6"
```
Comment (henrik-andreasson) [inline lang/938.english:20]: bad desc

File: lang/938.english [hunk 12]
```diff
@@ -1,81 +1,81 @@
...
+HELP[6]="Check space on device"
 
-DESCR[3]="(%s)"
-HELP[3]=""
+DESCR[7]="Desc 7"
```
Comment (henrik-andreasson) [inline lang/938.english:23]: bad desc

File: lang/938.english [hunk 13]
```diff
@@ -1,81 +1,81 @@
...
+HELP[7]="Check space on device"
 
-DESCR[3]="(%s)"
-HELP[3]=""
+DESCR[8]="(%s)"
```
Comment (henrik-andreasson) [inline lang/938.english:26]: ?

File: related-available/938_mariabackup.sh [hunk 14]
```diff
@@ -173,26 +176,65 @@ mariabackup_incremental_prepare() {
...
+ read dret
+ COUNT=1
+ while [ $COUNT -le $NUM ]
+ do
+  mariabackup_prepare_incrementel ${PREP_DIR}/${COUNT}_INC ${PREP_BASE}
+        echo "--------------------"
+  ((COUNT++))
+ done
+else
+ echo "WRONG value, bale out"
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:228]: bale? -> exiting

File: related-available/938_mariabackup.sh [hunk 15]
```diff
@@ -173,26 +176,65 @@ mariabackup_incremental_prepare() {
...
+
+PREP_PATH=$1 
+NUM=$(basename $1|cut -f1 -d"_")
+read dret
+PREP_TYPE=$(basename $1|sed 's/.*_//')
+
+if [ "${PREP_TYPE}" == "FULL" ] ; then
+ echo "Prepere Full backup  ${PREP_PATH} for restore,,"
+ mariabackup_prepare_full ${PREP_PATH}
+elif [ "${PREP_TYPE}" == "CUM" ] ; then
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:198]: use another word

File: related-available/938_mariabackup.sh [hunk 16]
```diff
@@ -173,26 +176,65 @@ mariabackup_incremental_prepare() {
...
+ PREP_BASE=${PREP_DIR}/FULL 
+ echo "Prepere Cumulative backup  ${PREP_PATH} for restore,,"
+ echo "Use Full backup from ${PREP_BASE}" 
+ mariabackup_prepare_full ${PREP_BASE}
+ echo "Continue to prep cumulativ backup: ${PREP_DIR}"
+ mariabackup_prepare_incrementel ${PREP_PATH} ${PREP_BASE}
+elif [ "${PREP_TYPE}" == "INC" ] ; then
+ PREP_DIR=$(dirname $PREP_PATH)
+ PREP_BASE=${PREP_DIR}/FULL 
+ echo "Prepere Incremetel backup  ${PREP_PATH} for restore,,"
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:209]: spelling incremental

File: related-available/938_mariabackup.sh [hunk 17]
```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
...
+    ret_restore=$($MARIABACKUP_BIN --copy-back --target-dir="${FULL_BACKUP_DIR}" --datadir=${MYSQL_DB_PATH} 2>&1)
     retcode2=$?
     DATEDONE2=$(date +"%s")
     let TIMETOCOMPLEATE2="$DATEDONE2 - $DATESTART2" || true # not to stop script
 
-    if [ $retcode2 -eq 0 ] ; then
-        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[18]} -d "${DESCR[18]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
-    else
+    if [ $retcode2 != 0 ] ; then
+    #    printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[18]} -d "${DESCR[18]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
```
Comment (henrik-andreasson) [inline related-available/938_mariabackup.sh:259]: remove commented code if done

File: related-available/938_mariabackup.sh (hunk 2 above)
Comment (henrik-andreasson) [reply related-available/938_mariabackup.sh:291]: also I do really not like menus in syscheck it is not possible to automate.
can we convert this to a two stage thing? 1. run with --list 2. run with --select

**Review by henrik-andreasson — APPROVED** [review]

**Review by henrik-andreasson — APPROVED** [review]

**Review by henrik-andreasson — APPROVED** [review]

**Comment by falleri** [comment]
Hi
 it is possible to run -r path_to_recovery_dir_yyyy-mm-dd
It can then be FULL, INC_5 or CUM_3
Is that ok.

Mats

tis 2 mars 2021 kl. 22:02 skrev henrik-andreasson <notifications@github.com

**Comment by falleri** [comment]
Oki

tis 2 mars 2021 kl. 22:00 skrev henrik-andreasson <notifications@github.com

**Comment by falleri** [comment]
Ok
I need to rember what i think short memorial
I fix
Mats

tis 2 mars 2021 kl. 21:54 skrev henrik-andreasson <notifications@github.com

**Comment by henrik-andreasson** [comment]
ok if we can run restore auto okok

**Comment by henrik-andreasson** [comment]
updates accepted

**Pull request merged**